### PR TITLE
WIP:Testing editable fields collec

### PR DESCRIPTION
**Pull request by DavidGeismarLtd**
@ManyBaby WIP but ou can already give me feedbacks and comments and I will try to implement them as I go.
This PR will have : 
- Ruby tests for the whole editable backend infrastructure
- JS test for the whole editable frontend infrastructure
- maybe a few capybara feature tests for the editable UI.

### CODE REVIEW

File: spec/lib/editable_fields_collection_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+RSpec.describe EditableFieldsCollection do
+
+  def create_categories
+    create :category, :dpc
+    create :category, :masterclass
+    create :category, :university
+  end
+
+  def create_editable_fields
+
```
Comment (ManyBaby) [inline spec/lib/editable_fields_collection_spec.rb:11]:
```suggestion
```

**Comment by ManyBaby** [comment]
🙏🏾  this PR is already big engough to handle. The smaller the better.

File: spec/lib/editable_fields_collection_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  describe ".new" do
+    describe 'takes a config of editables and builds a constant time accessible data structure from it' do
+      it 'creates top level keys with the name of the classes of the object given underscored' do
+        expect(fields.collection.keys).to match_array %w( editable_field category)
+      end
+
+      it 'creates second level keys based on the label of the objects given' do
+        expect(fields.collection["editable_field"].keys).to match_array EditableField.all.pluck(:label)
```
Comment (ManyBaby) [inline spec/lib/editable_fields_collection_spec.rb:34]: I think you should use a static value to evaluate here and make the code more explicit and evaluate the expected result.

File: spec/lib/editable_fields_collection_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        expect(fields.collection["category"].keys).to match_array Category.all.map(&:label)
```
Comment (ManyBaby) [inline spec/lib/editable_fields_collection_spec.rb:35]: same here

File: spec/lib/editable_fields_collection_spec.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        end
+      end
+    end
+  end
+
+  describe '[]' do
+    it 'it finds a specific dataEditable in the collection' do
+      category = Category.first
+      target = fields["category"][category.label]
+      expect(target.instance.attributes).to eq category.attributes
```
Comment (ManyBaby) [inline spec/lib/editable_fields_collection_spec.rb:53]: same here

File: spec/lib/editable_fields_collection_spec.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    it 'it fallsback to editable_field default first level key if it doesnt find the key' do
+      field = EditableField.first
+      target = fields[field.label]
+      expect(target.instance.attributes).to eq field.attributes
```
Comment (ManyBaby) [inline spec/lib/editable_fields_collection_spec.rb:59]: same here

**Review by ManyBaby — CHANGES_REQUESTED** [review]
The PR is almost good in itself. Thank you for setting up database cleaner.
I would like our tests to be explicit. That means that in the expectations we must avoid evalutate code with code and instead compare a code returned value to a static value whenever possible.

**Comment by DavidGeismarLtd** [comment]
> The PR is almost good in itself. Thank you for setting up database cleaner.
> I would like our tests to be explicit. That means that in the expectations we must avoid evalutate code with code and instead compare a code returned value to a static value whenever possible.

@ManyBaby understood
- I will try to rework it in a more "readable" manner.
- I will restrain this PR to only editable stuff (already pushed tests for data editable) 

I will handle the JS and feature tests on a separate PR

**Pull request merged**